llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -134,6 +181,18 @@ def gen_rotary_embedding(
                 device=
                 'cpu',  # FSDP does not materialize modules with meta buffers, hence device is set to cpu
             )
+        elif rope_hf_config['type'] == 'llama3':
+            llama_rope_config = {**rope_hf_config}
+            llama_rope_config['rope_type'] = rope_hf_config.get('type')
+            return LlamaRotaryEmbedding(
+                config=PartialLlamaConfig(
```

Comment:
so i'm understanding -- the new class takes in just the complete llama config right

Reply:
yessir